agent/engine/dependencygraph/graph.go

```diff
@@ -124,7 +131,7 @@ func DependenciesAreResolved(target *api.Container,
 	}
 
 	if !verifyStatusResolvable(target, nameMap, target.SteadyStateDependencies, onSteadyStateIsResolved) ||
-		!verifyTransitionDependenciesResolved(target, nameMap) {
+		!verifyTransitionDependenciesResolved(target, nameMap, resourcesMap) {
```

Comment:
You'd need `verifyTransitionDependenciesResolved` to return a `bool` and an `error`. Otherwise, both container and resource dependencies not being satisfied will be clubbed under `DependentContainerNotResolvedErr`, which would be misleading